python/ray/data/datastream.py

```diff
@@ -597,6 +597,11 @@ def map_batches(
 
         target_block_size = None
         if batch_size == "default":
+            ctx = DataContext.get_current()
+            if ctx.strict_mode:
+                raise ValueError(
+                    "`batch_size` must be specified when using strict " "mode "
+                )
```

Comment:
I just figured there're seveal APIs depending on default batch_size, e.g. [random_sample](https://github.com/ray-project/ray/blob/master/python/ray/data/datastream.py#L1137), add_column, drop_columns, select_columns etc. We need to handle them properly.

Reply:
Updated to still use a default for CPU `map_batches`